enhancements/bundle-validation.md

```diff
@@ -0,0 +1,222 @@
+
+- **Issue 2:** Rather than rewriting the courier project to provide static analysis for bundles provided as Catalog Images, the new library can be written and then imported by tools like _scorecard_ to perform validation.
+
+### Goals
+
+- Define a set of rules/validations so that bundles with their specific set of configurations can be validated by checking the required information and components.
+
+### Non-Goals
+
+- The validation compares different versions of bundles from the same package.
+ 
+- The validation considers the validity of a bundle in the context of an index, which should be part of [operator-registry](https://github.com/operator-framework/operator-registry).
+
+- The validation guarantees the successful installation of a bundle (an operator).
+
+## Proposal
+
+### User Stories
+
+As an Operator developer or maintainer, I would like to discover my missing vital metadata and ensure that my Operator bundle is well-formed.
+
+As a cluster administrator or release pipeline owner, I would want to be aware of malformed Operator bundles so that I can prevent them from being released into a catalog for cluster tenants to install.
+
+As a community contributor to the bundle validation library, I would want to understand the workflow of contributing to the core validation strategies or customized validations as part of an extended package.
+
+### Implementation Details/Notes/Constraints
+
+- The bundle validation library should list out errors based on static scanning of a bundle directory and locate their root causes if applicable. 
+
+- Much like project [flak8](https://pypi.org/project/flake8/) where a [list of validations](https://www.flake8rules.com/) can be individually enabled/disabled, individual bundle validating rule should also be separable. Ultimately, tools that are interested in importing the validation library should be able to cherry-pick validations used for different scenarios and configurations. 
+
+- The library should provide all validations currently available within the operator-courier.
+
+- The bundle validation library will be used to validate OLM/registry manifests and expose an interface to add validators for any scheme. 
+
+- Errors and alerting messages should be formatted in a standardized fashion.
+
+- The structure of the validation library needs to be well-organized and defined in a README.md to encourage community contribution.
+
+- The library should allow customized validation strategies as part of an extended package.
+
+- The library could potentially group different sets of validations to support specific bundle types or targeted cluster settings, ie: "validations for Kube 1.18", etc.
```

Comment:
Is this going to happen or is it a future change? 

`potentially group` makes me think we are not sure what will be implemented for this.

Reply:
it was originally thought to be future work, but I don't see why it can not be part of this enhancement.